src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -282,7 +309,6 @@ public void UseDevelopmentSettings()
         {
             Tracing.ConsoleLevel = TraceLevel.Verbose;
             Queues.MaxPollingInterval = TimeSpan.FromSeconds(2);
-            Singleton.ListenerLockPeriod = TimeSpan.FromSeconds(15);
```

Comment:
Note that I removed this restriction now. I think this was there to allow you to kill the process and restart without waiting a minute. Now that's not a concern. This will also keep things more consistent in Functions so debug mode won't shorten your timeout until the host restarts.